tests/queries/tests.py

```diff
@@ -3918,3 +3920,31 @@ def test_ticket_23622(self):
             set(Ticket23605A.objects.filter(qy).values_list('pk', flat=True))
         )
         self.assertSequenceEqual(Ticket23605A.objects.filter(qx), [a2])
+
+
+class Ticket29697Tests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.c1 = Ticket29697Company.objects.create(name="c1")
+        cls.c2 = Ticket29697Company.objects.create(name="c2")
+        cls.c3 = Ticket29697Company.objects.create(name="c3")
+        cls.c4 = Ticket29697Company.objects.create(name="c4")
+        cls.s1 = Ticket29697Status.objects.create(company=cls.c1, key="s-1-1")
+        cls.s2 = Ticket29697Status.objects.create(company=cls.c1, key="s-1-2")
+        cls.s3 = Ticket29697Status.objects.create(company=cls.c2, key="s-2-1")
+        cls.s4 = Ticket29697Status.objects.create(company=cls.c3, key="s-3-1")
+        cls.u1 = Ticket29697User.objects.create(company=cls.c1, is_superuser=True, id=4)
+        cls.u2 = Ticket29697User.objects.create(company=cls.c2, is_superuser=True)
+        cls.u3 = Ticket29697User.objects.create(company=cls.c3, is_superuser=True)
+        cls.u4 = Ticket29697User.objects.create(company=cls.c4, is_superuser=True)
+
+    def test_ticket_29697(self):
+        status = Ticket29697Status.objects.filter(
+            Q(company__employees__is_superuser=True) |
+            ~Q(company__employees__permissions__contains=Concat(Value('records.'), 'key', Value('.denied'))),
+            company__employees__pk=4).order_by('id')
```

Comment:
I'd be great if you could make the test case more concise somehow. I doubt all of these criterias are required to hit the issue and they add a lot of noise which makes future debugging harder.